File: app/src/main/java/com/tapan/weather/forecast/core/base/custom/CustomDialog.java
package com.tapan.weather.forecast.core.base.custom;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatTextView;
import com.tapan.weather.forecast.R;

public class CustomDialog extends Dialog {
  private AppCompatTextView mErrorMessage;

  public CustomDialog(@NonNull Context context) {
    super(context);
  }

  public CustomDialog(@NonNull Context context, int themeResId) {
    super(context, themeResId);
  }

  protected CustomDialog(@NonNull Context context, boolean cancelable,
      @Nullable OnCancelListener cancelListener) {
    super(context, cancelable, cancelListener);
  }

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.dialog_custom);
    mErrorMessage = findViewById(R.id.error_message);
  }

  public void setMessage(String message) {
    mErrorMessage.setText(message);
  }
}

File: app/src/main/java/com/tapan/weather/forecast/core/base/custom/CustomDialog.kt
package com.tapan.weather.forecast.core.base.custom

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.view.WindowManager
import androidx.appcompat.widget.AppCompatTextView
import com.tapan.weather.forecast.R

class CustomDialog(context: Context) : Dialog(context) {

  private var mErrorMessage: AppCompatTextView? = null

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    window?.setLayout(
        WindowManager.LayoutParams.MATCH_PARENT,
        WindowManager.LayoutParams.MATCH_PARENT
    )

    setContentView(R.layout.dialog_custom)
    mErrorMessage = findViewById(R.id.error_message)
  }

  fun setMessage(message: String?) {
    mErrorMessage?.text = message
  }
}